core/arch/aarchxx/mangle.c

```diff
@@ -2149,12 +2149,14 @@ restore_tls_base_to_stolen_reg(dcontext_t *dcontext, instrlist_t *ilist, instr_t
 {
     /* store app val back if it might be written  */
     if (instr_writes_to_reg(instr, dr_reg_stolen, DR_QUERY_INCLUDE_COND_DSTS)) {
-        PRE(ilist, next_instr,
-            XINST_CREATE_store(dcontext,
-                               opnd_create_base_disp(reg, REG_NULL, 0,
-                                                     os_tls_offset(TLS_REG_STOLEN_SLOT),
-                                                     OPSZ_PTR),
-                               opnd_create_reg(dr_reg_stolen)));
+        instr_t *store = XINST_CREATE_store(
+            dcontext,
+            opnd_create_base_disp(reg, REG_NULL, 0, os_tls_offset(TLS_REG_STOLEN_SLOT),
+                                  OPSZ_PTR),
+            opnd_create_reg(dr_reg_stolen));
+        instr_set_our_mangling(store, true);
```

Comment:
This is only used by routines inside d_r_mangle() so this should already be set. How about adding an assert on instrlist_get_our_mangling instead and commenting in the function comment that this is only callable inside d_r_mangle().

Reply:
See above.